crates/dpapi/src/rpc/auth.rs

```diff
@@ -0,0 +1,234 @@
+use sspi::builders::{AcquireCredentialsHandle, WithoutCredentialUse};
+use sspi::credssp::SspiContext;
+use sspi::{
+    AcquireCredentialsHandleResult, BufferType, ClientRequestFlags, CredentialUse, Credentials, CredentialsBuffers,
+    DataRepresentation, EncryptionFlags, SecurityBuffer, SecurityBufferFlags, SecurityBufferRef, SecurityStatus, Sspi,
+};
+use thiserror::Error;
+
+use crate::rpc::pdu::{AuthenticationLevel, SecurityProvider, SecurityTrailer};
+
+#[derive(Error, Debug)]
+pub enum AuthError {
+    #[error("{0} security provider is not supported")]
+    SecurityProviderNotSupported(&'static str),
+
+    #[error("SSPI authorization error: {0}")]
+    Sspi(#[from] sspi::Error),
+
+    #[error(transparent)]
+    IntConversion(#[from] std::num::TryFromIntError),
```

Comment:
suggestion: Note that at this level I think it’s more relevant to give a high-level meaning to the failed conversion.

Example: `BadSecurityTrailer`, without the from-conversion.

This `IntConversion` is convenient and straightforward, but does not tell much of the story.

Reply:
done